### ISSUE TITLE
unexpected directory layout

### ISSUE THREAD

**Issue by AlexTuan1024** [pierrre/imageserver#19]
## System Info
- OS: Windows10 
- Go version: go1.10.1 windows/amd64
- GOPATH: C:\projects
- GOROOT: C:\Go\
- DEP: dep 
- PROJECT NAME: image

## Expected Bahavior
Server started
## Actual Behavior
unexpected directory layout:
        import path: github.com/pierrre/imageserver
        root: C:\projects\src
        dir: C:\projects\src\image\vendor\github.com\pierrre\imageserver
        expand root: C:\projects\src
        expand dir: C:\projects\src\image\vendor\github.com\pierrre\imageserver
        separator: \
## Description
I tried your demo https://github.com/pierrre/imageserver/blob/master/examples/simple/simple.go with `dep` instead of `go get`, and this error happend when I tried to testrun.

This only happens when I use dep, and `imageserver` is under `vendor` folder.
It works when I use `go get`, but I don't see why it doesn't work anymore when I use `dep`.

**Comment by pierrre** [comment]
I don't have this issue on my side:
```
➜  imageserver git:(master) dep init
  Using master as constraint for direct dep github.com/pierrre/lrucache
  Locking in master (f5fef57) for direct dep github.com/pierrre/lrucache
  Locking in v1.1.0 (b4deda0) for transitive dep github.com/golang/protobuf
  Using master as constraint for direct dep github.com/golang/groupcache
  Locking in master (24b0969) for direct dep github.com/golang/groupcache
  Using master as constraint for direct dep github.com/nfnt/resize
  Locking in master (83c6a99) for direct dep github.com/nfnt/resize
  Using master as constraint for direct dep github.com/pierrre/githubhook
  Locking in master (3acdf47) for direct dep github.com/pierrre/githubhook
  Using master as constraint for direct dep github.com/pierrre/compare
  Locking in master (083aa15) for direct dep github.com/pierrre/compare
  Using master as constraint for direct dep github.com/bradfitz/gomemcache
  Locking in master (1952afa) for direct dep github.com/bradfitz/gomemcache
  Using ^1.1.2 as constraint for direct dep github.com/disintegration/gift
  Locking in v1.1.2 (6844aa0) for direct dep github.com/disintegration/gift
  Using master as constraint for direct dep golang.org/x/image
  Locking in master (af66def) for direct dep golang.org/x/image
  Using master as constraint for direct dep github.com/pierrre/imageutil
  Locking in master (dd32da5) for direct dep github.com/pierrre/imageutil
  Using ^6.11.0 as constraint for direct dep github.com/go-redis/redis
  Locking in v6.11.0 (0f9028a) for direct dep github.com/go-redis/redis

➜  imageserver git:(master) ✗ cd examples/simple

➜  simple git:(master) ✗ go run -v simple.go
github.com/pierrre/imageserver/vendor/github.com/pierrre/imageutil
github.com/pierrre/imageserver/vendor/github.com/disintegration/gift
github.com/pierrre/imageserver
github.com/pierrre/imageserver/source
github.com/pierrre/imageserver/image
github.com/pierrre/imageserver/image/internal
github.com/pierrre/imageserver/http
github.com/pierrre/imageserver/testdata
github.com/pierrre/imageserver/image/gif
github.com/pierrre/imageserver/image/jpeg
github.com/pierrre/imageserver/image/png
github.com/pierrre/imageserver/http/image
github.com/pierrre/imageserver/http/gift
github.com/pierrre/imageserver/image/gift
command-line-arguments
```



> This only happens when I use dep, and imageserver is under vendor folder.

I'm not sure this is recommended.
My project is cloned here:
```
➜  imageserver git:(master) ✗ pwd
/home/pierre/go/src/github.com/pierrre/imageserver
```
and my GOPATH is `/home/pierre/go`

This directory structure `C:\projects\src\image\vendor\github.com\pierrre\imageserver` is very strange.
I'm not sure this is the best practice.

**Comment by AlexTuan1024** [comment]
It seems like this problem was caused by my weird directory structure.

I tried to make it this way: `C:\projects\src\my-gitlab.xxxxx.net\xxxx\image`, ran `dep init` again and the result was the same as yours.

The tricky part is, I always put my projects in the %GOPATH%\src folder such as `C:\projects\src\myproject` and everything was fine. Now I feel confused.

BTW I saw similar issues submitted to different golang repositories. Maybe the same reason as mine.

Feel free to close this issue. Thank you so much, Pierre.
